README.md

```diff
@@ -182,6 +182,69 @@ You can intermittently poll whether the operation has finished by using the `Pol
   }
 ```
 
+#### `Azure::Core::Context`
+
+Most Azure SDK Service Client methods accept an optional `Azure::Core::Context` parameter, which is used to enable cancellation of the operation or to
+establish an absolute deadline for the operation.
+
+This is useful when you want to assign a time limit on an operation to ensure that it completes in a "reasonable" timeframe. For instance, the 
+snippet below will cancel a blob client upload after 5 seconds.
+
+<!-- @insert_snippet: CreateBlobContext -->
+```cpp
+      Azure::Core::Context cancelledIn5s = Azure::Core::Context::CreateWithDeadline(
+          std::chrono::system_clock::now() + std::chrono::seconds(5));
+
+      auto containerClient = BlobContainerClient::CreateFromConnectionString(
+          GetConnectionString(), containerName + std::to_string(i));
+      containerClient.CreateIfNotExists({}, cancelledIn5s);
+      for (int j = 0; j < 3; ++j)
+      {
+        BlockBlobClient blobClient
+            = containerClient.GetBlockBlobClient(blobName + std::to_string(j));
+        blobClient.UploadFrom(
+            reinterpret_cast<const uint8_t*>(blobContent.data()),
+            blobContent.size(),
+            {},
+            cancelledIn5s);
+      }
+```
+
+`Context` objects can also be directly cancelled using the `Cancel()` method.
+
+`Context` objects form a directed tree, where a child context can be created from a parent context. 
+The context tree is unidirectional and acyclic.
+
+These are the basic operations that can be performed on a `Context` object:
+
+* Create a child context from a parent context with a Key/Value pair. This is useful for associating metadata with a context.
+* Create a child context from a parent context with a deadline. This is useful for setting a timeout.
+* Cancel a context. This will cancel the context and all its children. Note that there is no way of un-cancelling a context.
+* Check if a context is cancelled.
+
+When a context is copied from another context, the copied context will share state with the original context. 
+This means that if the original context is cancelled, the copied context will also be cancelled.
+
+Cancellation of a `Context` is a permanent operation. Once a context is cancelled, it cannot be un-cancelled.
+
+When a `Context` is cancelled, it is typically indicated by throwing `Azure::Core::OperationCancelledException`. This exception can be caught and handled by the application.
```

Comment:
https://github.com/Azure/azure-sdk-for-cpp/pull/5676#discussion_r1654054423
Bringing up an unresolved comment (it was hidden in the previous comments):

nit: The wording makes it seem like calling Cancel on Context will throw a `OperationCancelledException`

```suggestion
When a client operation fails with an `Azure::Core::OperationCancelledException`, it is typically due to a `Context` getting cancelled. This exception can be caught and handled by the application.
```